lib/hutch/waiter.rb

```diff
@@ -14,10 +18,20 @@ def wait_until_signaled
       self.sig_read, self.sig_write = IO.pipe
 
       register_signal_handlers
-      wait_for_signal
-
-      sig = sig_read.gets.strip.downcase
-      logger.info "caught sig#{sig}, stopping hutch..."
+      
+      begin
+        wait_for_signal
+
+        sig = sig_read.gets.strip.downcase
+        if USER_SIGNALS.include?(sig.upcase)
+          logger.info "FUN TIME! #{sig}"
```

Comment:
Ah, I should have written "This is a handler for signal SIGUSR1, the behavior of which we will define as a community".

The current implementation is only a skeleton logging "proof" that it works.